source/GLaDOS.cpp

```diff
@@ -179,6 +180,18 @@ void GLaDOS::update(float dt) {
 			break;
 		}
 	}
+
+	if (rand() % 15 == 1 && !ship->getChallenge()) {
+		((int)(rand() % 2 == 0)) ? ship->setRollDir(0) : ship->setRollDir(1);
+		int p = (int)(rand() % ship->getDonuts().size());
```

Comment:
Actually, no, we also need to check that this player is active.

Reply:
updated